src/jdk.hotspot.agent/macosx/native/libsaproc/MacosxDebuggerLocal.m

```diff
@@ -50,8 +50,14 @@
 #define amd64 1
 #endif
 
-#if amd64
+#if defined(__arm64__) && !defined(aarch64)
+#define aarch64 1
```

Comment:
Cool, I've opened https://bugs.openjdk.java.net/browse/JDK-8254941 for SA implementation